### DISCUSSION TITLE
Is it possible to generate rust files from proto files without using `tonic_build`?

### DISCUSSION THREAD

**Discussion by frederikhors** [hyperium/tonic#1387] [Q&A]
Is it possible to generate rust files from proto files without using `tonic_build`?

I can open a PR to improve docs about this.

Thanks in advance.

**Comment by wez** [comment]
Did you figure out the answer to this? I'm not enthusiastic about installing the protobuf compiler on a wide variety of systems and would much prefer to pre-generate and check this artifact in to the repo.

**Comment by 0xcaff** [comment]
https://github.com/hyperium/tonic/discussions/1092
